osu.Game.Rulesets.Taiko/Difficulty/TaikoDifficultyCalculator.cs

```diff
@@ -84,6 +84,10 @@ protected override DifficultyAttributes CreateDifficultyAttributes(IBeatmap beat
             double starRating = 1.4 * separatedRating + 0.5 * combinedRating;
             starRating = rescale(starRating);
 
+            // For the time being, we use the clockrate at the beginning of the map for OD and AR attributes
+            double baseClockRate = 1;
+            baseClockRate = mods.OfType<IApplicableToRate>().Aggregate(1.0, (prod, mod) => prod * mod.ApplyToRate(prod));
```

Comment:
Note that this simplification means that only the initial rate will have an impact on the map. For instance, consider Wind Up with rate going 0.5x -> 2.0x, and Wind Down with rate going 2.0x -> 0.5x. While the start/end rate values are the same but swapped, the Wind Down hitwindow will be twice as short as normally, while Wind Up's will be twice as long and therefore considered to be easier (and Wind Up is arguably *more* difficult in this scenario, as you have to play the final 25% of the map on the final 2.0x rate).

A reasonable compromise may be to expose some sort of "minimum"/"maximum" rate that a mod can return, and use either here.

Reply:
right, this was already something I was concerned about. I would roll with the "minimum" rate out of those two options, so that difficulty calculation runs off of a worst-case-scenario. an additional compromise I was considering was using the mid-point (average?) rate across the whole map, and I haven't thought of any relevant complications there yet.

definitely something I want to reconsider before I'm fully happy with this